src/rpc/error.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::chain::store::Error as ChainError;
+use crate::key_management::Error as KeyManagementError;
+use crate::libp2p::ParseError;
+use crate::message_pool::Error as MessagePoolError;
+use crate::state_manager::Error as StateManagerError;
+
+use jsonrpsee::types::error::{ErrorObjectOwned, INTERNAL_ERROR_CODE};
+
+pub struct JsonRpcError {
+    error: ErrorObjectOwned,
+}
+
+impl From<anyhow::Error> for JsonRpcError {
```

Comment:
I don't think `INTERNAL_ERROR_CODE` is suitable for all of these - that's basically a HTTP 500

Reply:
`-32603` is for an internal error no? According to the JSON-RPC spec here: 

https://www.jsonrpc.org/specification#error_object

Reply:
If you have no way to differentiate between errors - internal error seems fine. Not sure what else can be done here. There's a `SERVER_ERROR`, but that's a range of codes. And an `INTERNAL_ERROR`. Since we don't really know what to do with each individual error - I guess the only thing left here is to group them all and treat as the same error code. I can't see any other error code suitable for this purpose.

Reply:
Ah, I now understand Aatif's comment better. We have many different error types, but all of them are treated as `INTERNAL_ERROR`.

I'd say that perhaps `anyhow::Error` is actually suitable for this, but then again we have another bunch of types that we could at least differentiate. 

I guess we could use `Server error` instead with a bunch of custom-defined codes per error type in `From` definition.

Reply:
`INTERNAL_ERROR` is suitable for e.g handler panic - treat it like HTTP `500 Internal Server Error `.

Use other error codes (outside of the -32000..=-32099 range), here.

OTOH, if lotus always just always returns `INTERNAL_ERROR`, we might decide to copy that.
After this discussion, I'll leave this up to your judgement

Reply:
I'm ok with improving error codes but in a subsequent PR. Let's finish this one to avoid dealing with more merge conflicts.